app/CocoaPods/CPUserProject.m

```diff
@@ -85,6 +126,47 @@ - (void)textDidChange:(NSNotification *)notification;
   NSTextView *textView = notification.object;
   NSString *contents = textView.string;
   self.contents = contents;
+
+  [NSObject cancelPreviousPerformRequestsWithTarget:self selector:@selector(parsePodfile) object:nil];
+  [self performSelector:@selector(parsePodfile) withObject:nil afterDelay:1];
+}
+
+- (void)parsePodfile;
+{
+  [RBObject performBlock:^{
+    RBPathname *pathname = [RBObjectFromString(@"Pathname") new:self.fileURL.path];
+
+    @try {
+      RBPodfile *podfile = [RBObjectFromString(@"Pod::Podfile") from_ruby:pathname :self.contents];
+      NSArray *plugins = podfile.plugins.allKeys;
+      dispatch_async(dispatch_get_main_queue(), ^{
+        NSLog(@"Podfile Plugins: %@", plugins);
+        self.podfilePlugins = plugins;
+        // Clear any previous syntax errors.
+        self.editor.syntaxErrors = nil;
+      });
+    }
+
+    @catch (NSException *exception) {
+      // In case of a Pod::DSLError, try to create a UI syntax error out of it.
```

Comment:
If you were referring to using exceptions in general, then no, afaik Foundation classes should be safely released, especially when doing the autorelease pools right (that is, around the complete try/catch clause).

I believe it’s AppKit where Apple doesn’t make guarantees about classes correctly cleaning up after exceptions.